src/server/tests/test_handlers.py

```diff
@@ -0,0 +1,171 @@
+        self.dal = DalMock()
+        handlers.DAL = self.dal
+
+    def test_home_without_user_render_login(self):
+        response = self.app.get('/', follow_redirects=True)
+        self._assert_is_login(response)
+
+    def test_home_with_user_render_docs(self):
+        self._login()
+        response = self.app.get('/', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_login_with_user_connected_redirect_home(self):
+        self._login()
+        response = self.app.get('/login', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_login_post_valid_redirect_home(self):
+        response = self.app.post('/login', data=dict(email='mark', password='dadada'), follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_login_post_invalid_redirect_login(self):
+        response = self.app.post('/login', data=dict(email='mark', password='JoIsNoOne'), follow_redirects=True)
+        self._assert_is_login(response)
+        self.assertTrue('Unknown email or invalid password' in response.data)
+
+    def test_disconnect_disconnect_and_redirect_login(self):
+        self._login()
+        response = self.app.get('/disconnect', follow_redirects=True)
+        with self.app.session_transaction() as session:
+            self.assertFalse('email' in session)
+        self._assert_is_login(response)
+
+    def test_disconnect_not_connected_redirect_login(self):
+        response = self.app.get('/disconnect', follow_redirects=True)
+        self._assert_is_login(response)
+
+    def test_register_get_without_user_display_register(self):
+        response = self.app.get('/register', follow_redirects=True)
+        self._assert_is_register(response)
+
+    def test_register_get_with_user_redirect_home(self):
+        self._login()
+        response = self.app.get('/register', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_register_post_with_user_redirect_home(self):
+        self._login()
+        response = self.app.post('/register', follow_redirects=True)
+        self._assert_is_home(response)
+
+    def test_register_post_with_existing_email(self):
+        response = self.app.post('/register', data=dict(email='mark', password='', interests=''), follow_redirects=True)
+        self._assert_is_register(response)
+        self.assertTrue('This account already exists' in response.data)
+
+    def test_register_post_with_new_email_save_profile_redirect_login(self):
+        post_data = dict(email='alan', password='alan', interests='rockets\r\ncars')
+        response = self.app.post('/register', data=post_data, follow_redirects=True)
+        self.dal.saved_password = pswd.hash_password('alan')
+        self.assertEquals('alan', self.dal.saved_user.email)
+        self.assertEquals(['rockets', 'cars'], self.dal.saved_user.interests)
+        self.assertEquals(1, len(self.dal.computed_user_profiles))
+        self._assert_is_login(response)
```

Comment:
We should log in the user and redirect to the home page instead of redirecting to the login page. It's obvious, but before modifying this test we must modify the register handler.